packages/typespec-go/src/emitter.ts

```diff
@@ -28,10 +28,56 @@ import { EmitContext, NoTarget } from '@typespec/compiler';
 import 'source-map-support/register.js';
 import { reportDiagnostic } from './lib.js';
 import { CodegenError } from '../../codegen.go/src/errors.js';
+import { execSync } from 'child_process';
 
 export async function $onEmit(context: EmitContext<GoEmitterOptions>) {
   try {
     await generate(context);
+
+    // probe to see if Go tools are on the path
+    try {
+      execSync('go version', { encoding: 'ascii' });
+    } catch {
+      context.program.reportDiagnostic({
+        code: 'GoVersion',
+        severity: 'warning',
+        message: 'skip executing post emitter steps (is go on the path?)',
+        target: NoTarget,
+      });
+
+      // no Go tools available so exit
+      return;
+    }
+
+    try {
```

Comment:
Would be better to have the format command as the last to run. Sometimes transforms can mess with the code, so the formatter needs to be run after the transforms.